### PR TITLE
Limit DropDownLists popup height if the container is larger than a window (T859133)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Review by DokaRus — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js [hunk 1]
```diff
@@ -1077,6 +1077,28 @@ QUnit.module('popup', moduleConfig, () => {
...
     });
 
+    QUnit.test('popup max height are limited by container bounds and window', function(assert) {
+        const items = [];
+        for(let i = 0; i < 100; i++) {
+            items.push(`item ${i}`);
+        }
+
+        const windowHeight = $(window).outerHeight();
+        const parentContainer = $('<div>').attr('id', 'specific-container').css('overflow', 'hidden').height(windowHeight * 2).appendTo('#qunit-fixture');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js]:
```suggestion
        const parentContainer = $('<div>')
            .attr('id', 'specific-container')
            .css('overflow', 'hidden')
            .height(windowHeight * 2)
            .appendTo('#qunit-fixture');
```

?

File: testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js [hunk 2]
```diff
@@ -1077,6 +1077,28 @@ QUnit.module('popup', moduleConfig, () => {
         parentContainer.remove();
     });
 
+    QUnit.test('popup max height are limited by container bounds and window', function(assert) {
+        const items = [];
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js:1081]:
```suggestion
        const items = Array.from(Array(100), (_, i) => `item ${i}`)
```
?

File: testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js [hunk 3]
```diff
@@ -1077,6 +1077,28 @@ QUnit.module('popup', moduleConfig, () => {
...
+
+        const instance = $('#dropDownList').dxDropDownList({
+            items,
+            dropDownOptions: {
+                container: parentContainer
+            },
+            opened: true
+        }).dxDropDownList('instance');
+
+        assert.roughEqual($(instance.content('.dx-overlay-content')).parent().height(), windowHeight / 2, 2, 'popup sizes are limited by window if overflow:hidden container is larger than window');
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js]: Extract `$(instance.content('.dx-overlay-content')).parent()` to const?

**Review by San4es — DISMISSED** [review]
LGTM

File: testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js (hunk 2 above)
Comment (DokaRus) [reply testing/tests/DevExpress.ui.widgets.editors/dropDownList.tests.js:1081]: Incompatible with IE11

**Review by San4es — APPROVED** [review]

**Pull request merged**